lib/iris/coords.py

```diff
@@ -1762,30 +1761,42 @@ def data(self, data):
         if data is None:
             raise ValueError('The data payload of a CellMeasure may not be '
                              'None; it must be a numpy array or equivalent.')
+        if is_lazy_data(data) and data.dtype.kind in 'iu':
+            # Disallow lazy integer data.  Integer values would be very
+            # unexpected anyway, but this avoids the *even more obscure*
+            # possibility that there are masked values, because masked integer
+            # data would cause dask to return data with the wrong dtype.
+            # This can all be fixed, as for cube data, by adding fill_value and
+            # dtype support, but for now we prefer just not to support this.
```

Comment:
And we can add this as a feature request at a later date once we have a concrete example of this being needed :+1: